src/dist/dist.rs

```diff
@@ -464,9 +464,16 @@ impl ToolchainDesc {
         }
     }
 
+    /// Toolchain channels are considered 'tracking' if it is one of the named channels
+    /// such as `stable`, or is an incomplete version such as `1.48`, and the
+    /// date field is empty.
     pub fn is_tracking(&self) -> bool {
         let channels = ["nightly", "beta", "stable"];
-        channels.iter().any(|x| *x == self.channel) && self.date.is_none()
+        lazy_static! {
+            static ref TRACKING_VERSION: Regex = Regex::new(r"^\d{1}\.\d{1,3}$").unwrap();
```

Comment:
Hmm, I guess the minor version needs 3 digits for when Rust gets a `1.100` release. Can you add that as a comment?

I think 3 digits is fine - back of the envelope math says we'd need 100 years of rust releases, which seems like more than far enough away.

Reply:
We've been 3 digits in minor for a while now.  If we have to change it, it's not hard :D